File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.cpp
class Solution {
public:
    long long maximumHappinessSum(std::vector<int>& happiness, int k) {
        std::sort(happiness.begin(), happiness.end(), std::greater<int>());
        long long ans = 0;
        for (int i = 0; i < k; i++) {
            ans += std::max(static_cast<long long>(happiness[i]) - i, 0LL);
        }
        return ans;
    }
};

File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        Arrays.sort(happiness);
        long ans = 0;
        for (int i = happiness.length - 1; i >= happiness.length - k; i--) {
            ans += Math.max(happiness[i] - (happiness.length - 1 - i), 0);
        }
        return ans;
    }
}